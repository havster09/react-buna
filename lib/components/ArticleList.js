import React from 'react';
import PropTypes from 'prop-types';

import ArticleContainer from './Article';

class ArticleList extends React.PureComponent {
  render() {
    return (
      <div>
        {Object.values(this.props.articles).map(article =>
          <ArticleContainer
            key={article.id}
            article={article}
            author={this.props.articleActions.lookupAuthor(article.authorId)}
          />
        )}
        <pre>
          {JSON.stringify(this.props.articles, null, 4)}
        </pre>
      </div>
    );
  }
}

ArticleList.propTypes = {
  articles: PropTypes.object,
  articleActions: PropTypes.object
};

export default ArticleList;
