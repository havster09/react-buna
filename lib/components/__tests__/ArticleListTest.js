import React from 'react';
import 'babel-polyfill';
import ArticleList from '../ArticleList';

import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe('ArticleList', () => {
  const testProps = {
    articles: {
      a: { id: 'a', date: 'Fri Jul 22 2016 00:00:00 GMT+0000 (UTC)' },
      b: { id: 'b', date: 'Fri Jul 22 2016 00:00:00 GMT+0000 (UTC)' },
      c: { id: 'c', date: 'Fri Jul 22 2016 00:00:00 GMT+0000 (UTC)' }
    },
    articleActions: {
      lookupAuthor: jest.fn(() => ({}))
    }
  };

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ArticleList
          articles={testProps.articles}
          articleActions={testProps.articleActions}
        />
      )
      .toJSON();
    expect(tree.children.length).toBe(4);
    expect(tree).toMatchSnapshot();
  });
});

describe('ArticleList Shallow test', () => {
  it('passes a shallow test', () => {
    const testProps = {
      articles: {
        a: { id: 'a', date: 'Fri Jul 22 2016 00:00:00 GMT+0000 (UTC)' },
        b: { id: 'b', date: 'Fri Jul 22 2016 00:00:00 GMT+0000 (UTC)' },
        c: { id: 'c', date: 'Fri Jul 22 2016 00:00:00 GMT+0000 (UTC)' }
      },
      articleActions: {
        lookupAuthor: jest.fn(() => ({}))
      }
    };
    const wrapper = shallow(
        <ArticleList
          {...testProps}
        />
      );
    expect(wrapper.find('ArticleContainerHOC').length).toBe(3);
  });
});
