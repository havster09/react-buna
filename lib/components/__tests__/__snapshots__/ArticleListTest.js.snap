// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
<div>
  <div
    style={
      Object {
        "borderBottomColor": "#aaa",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "marginBottom": 10,
        "paddingBottom": 10,
      }
    }
  >
    <div
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    />
    <div
      style={
        Object {
          "color": "#888",
          "fontSize": "0.8em",
        }
      }
    >
      Fri Jul 22 2016
    </div>
    <div
      style={
        Object {
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <a
        href={undefined}
      >
         
      </a>
    </div>
    <div
      style={
        Object {
          "paddingLeft": 20,
        }
      }
    />
    <h1 />
    <h1 />
    <h1>
      veteran
    </h1>
    <h1>
      rifle_man
    </h1>
    <h1>
      [
          "Browning SLP",
          "Steyer"
      ]
    </h1>
  </div>
  <div
    style={
      Object {
        "borderBottomColor": "#aaa",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "marginBottom": 10,
        "paddingBottom": 10,
      }
    }
  >
    <div
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    />
    <div
      style={
        Object {
          "color": "#888",
          "fontSize": "0.8em",
        }
      }
    >
      Fri Jul 22 2016
    </div>
    <div
      style={
        Object {
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <a
        href={undefined}
      >
         
      </a>
    </div>
    <div
      style={
        Object {
          "paddingLeft": 20,
        }
      }
    />
    <h1 />
    <h1 />
    <h1>
      veteran
    </h1>
    <h1>
      rifle_man
    </h1>
    <h1>
      [
          "Browning SLP",
          "Steyer"
      ]
    </h1>
  </div>
  <div
    style={
      Object {
        "borderBottomColor": "#aaa",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "marginBottom": 10,
        "paddingBottom": 10,
      }
    }
  >
    <div
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    />
    <div
      style={
        Object {
          "color": "#888",
          "fontSize": "0.8em",
        }
      }
    >
      Fri Jul 22 2016
    </div>
    <div
      style={
        Object {
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <a
        href={undefined}
      >
         
      </a>
    </div>
    <div
      style={
        Object {
          "paddingLeft": 20,
        }
      }
    />
    <h1 />
    <h1 />
    <h1>
      veteran
    </h1>
    <h1>
      rifle_man
    </h1>
    <h1>
      [
          "Browning SLP",
          "Steyer"
      ]
    </h1>
  </div>
  <pre>
    {
        "a": {
            "id": "a",
            "date": "Fri Jul 22 2016 00:00:00 GMT+0000 (UTC)"
        },
        "b": {
            "id": "b",
            "date": "Fri Jul 22 2016 00:00:00 GMT+0000 (UTC)"
        },
        "c": {
            "id": "c",
            "date": "Fri Jul 22 2016 00:00:00 GMT+0000 (UTC)"
        }
    }
  </pre>
</div>
`;
